windows/src/desktop/kmconfig/Keyman.System.KMConfigMain.pas

```diff
@@ -0,0 +1,333 @@
+    class function Export(const Filename: string): Boolean; static;
+    class function Reset(const ID: string): Boolean; static;
+    class function SetValue(const ID, Value: string): Boolean; static;
+    class function Show(const P2: string): Boolean; static;
+    class procedure Log(const message: string);
+    class procedure LogError(const message: string);
+    class procedure LogWarning(const message: string);
+  public
+    class constructor ClassCreate;
+    class destructor ClassDestroy;
+    class procedure Run; static;
+  end;
+
+implementation
+
+uses
+  System.SysUtils,
+  System.Win.Registry,
+  Winapi.Windows,
+
+  Keyman.System.SettingsManager,
+  Keyman.System.SettingsManagerFile,
+  RegistryKeys;
+
+type
+  TANSIColor = record
+    Green, Grey, Default, Yellow, Red, White: string;
+  end;
+
+var
+  ANSIColor: TANSIColor;
+
+class constructor TKMConfig.ClassCreate;
+var
+  r: TRegistry;
+begin
+  inherited;
+  r := TRegistry.Create;
+  try
+    r.RootKey := HKEY_LOCAL_MACHINE;
+    IsAdmin := r.OpenKey(SRegKey_KeymanEngine_LM, True);
+  finally
+    r.Free;
+  end;
+
+  Settings := TKeymanSettings.Create;
+end;
+
+class destructor TKMConfig.ClassDestroy;
+begin
+  Settings.Free;
+  inherited;
+end;
+
+class procedure TKMConfig.Run;
+var
+  Command: string;
+  Result: Boolean;
+begin
+  Settings := TKeymanSettings.Create;
+  Command := ParamStr(1).ToLower;
+  if Command = 'import' then Result := Import(ParamStr(2))
+  else if Command = 'export' then Result := Export(ParamStr(2))
+  else if Command = 'reset' then Result := Reset(ParamStr(2))
+  else if Command = 'set' then Result := SetValue(ParamStr(2), ParamStr(3))
+  else if Command = 'show' then Result := Show(ParamStr(2))
+  else
+  begin
+    Result := False;
+    writeln('Usage: kmconfig command [options]');
+    writeln('Commands:');
+    writeln('  import <settings.json>   Imports settings into local machine registry');
+    writeln('  export <settings.json>   Exports settings from local machine registry to file');
+    writeln('  reset [id]               Resets one or all settings to default');
+    writeln('  set id value             Sets setting identified by id to value');
+    writeln('  show [-a] [id]           Lists one or all settings; -a to show all, even empty settings');
+    writeln('Note: registry changes may require elevation to succeed.');
+  end;
+
+  if Result
+    then ExitCode := 0
+    else ExitCode := 1;
+end;
+
+class procedure TKMConfig.Log(const message: string);
+begin
+  writeln(message);
+end;
+
+class procedure TKMConfig.LogError(const message: string);
+begin
+  Log(AnsiColor.Red+'ERROR: '+message+AnsiColor.Default);
+end;
+
+class procedure TKMConfig.LogWarning(const message: string);
+begin
+  Log(AnsiColor.Yellow+'WARNING: '+message+AnsiColor.Default);
+end;
+
+class function TKMConfig.Export(const Filename: string): Boolean;
+begin
+  TKeymanSettingsManager.Load(Settings);
+  TKeymanSettingsManagerFile.Export(Settings, Filename);
```

Comment:
I think I asked on the other PR if it's possible to return False if the Import/Export fails

Reply:
An exception will be raised if the Import/Export fails and a message written to console.